### ISSUE TITLE
RAND_DRBG_USED_FLAGS is problematic for ABI compat

### ISSUE THREAD

**Issue by mattcaswell** [openssl/openssl#7182]
The new DRBG API includes a define for `RAND_DRBG_USED_FLAGS` in the public header file. This is problematic for ABI compatibility. It seems to imply that we can't add new flags in ABI compat releases. If we do applications compiled against a previous release will not be using the correct value if linking at run time against a newer OpenSSL version with a different value for this define.

https://github.com/openssl/openssl/blob/a4a90a8a3bdcb9336b5c9c15da419e99a87bc6ed/include/openssl/rand_drbg.h#L20-L23

**Comment by mattcaswell** [comment]
Note that PR #6779 wants to add new flags.

**Comment by kroeckx** [comment]
How do we feel about removing it in the next release?

**Comment by kroeckx** [comment]
The purpose of the flag is to give an error in case something is build against a new version that knows the flag but run against an old version that doesn't know the flag. I think it's great that such a check is there.

But that define really seems to be for internal use only, so that we can give that error. External users are nothing with it.

**Comment by mspncp** [comment]
> How do we feel about removing it in the next release?
> ...
> But that define really seems to be for internal use only...

If I understand you right, you just mean to move it from the public headers to a private one or to drbg_lib.c. In this case, I agree with you.

**Comment by kroeckx** [comment]
Yes, that's what I'm saying.

**Comment by mattcaswell** [comment]
In which case, why don't we define it in the internal header now (give it a slightly different name to avoid a name clash). Then update the public header to basically say "don't use this define" - so that people don't start using it expecting it to be useful.

**Comment by mspncp** [comment]
See #7190.  It contains only the deprecation warning, Adding a second internal define is not necessary IMO.

**Comment by mspncp** [comment]
New commit message

``` 
    Replace the public RAND_DRBG_USED_FLAGS #define by an internal constant

    The new DRBG API added the aforementioned #define. However, it
    is used internally only and having it defined publicly does not
    serve any purpose except causing potential version compatibilty
    problems.

    Fixes #7182
```
